### PR TITLE
app creation scripts update

### PR DESCRIPTION
**Pull request by derisen**
This PR updates the following files:
- apps.json: changed reply url type to 'Spa' in anticipation of upcoming portal changes
- sample.json: changed the replacement method and destination

new file(s):
- app-config.json: registration scripts have trouble with updating .ts files. Because of that, configuration parameters are moved to this .json file and loaded into app.module.ts

### CODE REVIEW

**Comment by jasonnutter** [comment]
> registration scripts have trouble with updating .ts files

Can you expand on this? What issues are there?

File: AppCreationScripts/apps.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
         "replyUrlsWithType": [
           {
             "url": "http://localhost:4200/",
-            "type": "Web"
+            "type": "Spa"
```
Comment (jasonnutter) [inline AppCreationScripts/apps.json]: This is unrelated to the other changes, and would prefer it be made separately when we are ready to update Angular to the auth code flow.

**Review by jasonnutter — CHANGES_REQUESTED** [review]
We're going to need to test this with the Azure Portal quickstart generation, which relies on these same replacements.

**Comment by derisen** [comment]
> > registration scripts have trouble with updating .ts files
> 
> Can you expand on this? Which scripts are these?

Sure -these are the powershell scripts in the `AppCreationScripts` folder. They automate the registration on AAD portal and replace the msal configuration parameters in the source code afterwards (my team uses them at live demos..) But I realized they mess up the configs in app.module.ts, in particular arrays (e.g `ProtectedResourceMap`). So instead I moved the parameters to `app-config.json`.

File: AppCreationScripts/apps.json (hunk 1 above)
Comment (derisen) [reply AppCreationScripts/apps.json]: I see. I'll revert this back.
